### PR TITLE
coreneuron: Bumps to v8.2.1

### PR DESCRIPTION
**Pull request by sergiorg-hpc**

### CODE REVIEW

**Review by olupton — APPROVED** [review]

**Review by pramodk — DISMISSED** [review]

**Review by matz-e — DISMISSED** [review]

**Review by alexsavulescu — DISMISSED** [review]

**Review by olupton — CHANGES_REQUESTED** [review]
Actually
```python
conflicts('+shared', when='@:8.2.0 +gpu')
```
is not correct :/

**Comment by sergiorg-hpc** [comment]
> 

Is it because of the version? If so, perhaps this simply could work for now, what do you think?:
```
conflicts('+shared', when='+gpu')
```

**Comment by olupton** [comment]
We want to allow `coreneuron@develop +gpu+shared` for use in CI pipelines.
I don't think this will be supported in any pre-9.0 release, so we could change the upper end of the range...but perhaps this conflict is doing more harm than good, seeing as pre-9.0 versions will just silently ignore +shared anyway when +gpu is passed... (cc: @iomaganaris )

**Comment by iomaganaris** [comment]
> We want to allow `coreneuron@develop +gpu+shared` for use in CI pipelines. I don't think this will be supported in any pre-9.0 release, so we could change the upper end of the range...but perhaps this conflict is doing more harm than good, seeing as pre-9.0 versions will just silently ignore +shared anyway when +gpu is passed... (cc: @iomaganaris )

Since the PR related to `+gpu+shared` has been already merged, I expect the next `CoreNEURON` tag to be based on top of it, so maybe:
```
conflicts('+shared', when='@:8.2.1 +gpu')
```
is enough?

**Comment by sergiorg-hpc** [comment]
> Since the PR related to `+gpu+shared` has been already merged, I expect the next `CoreNEURON` tag to be based on top of it, so maybe:
> 
> ```
> conflicts('+shared', when='@:8.2.1 +gpu')
> ```
> 
> is enough?

I have the suspicion that this will be a problem if, for some reason, there is a 8.2.2 release or similar. Based on @olupton's comment, I just changed it to be `@:8.9.9` to prevent further issues with this.

Sounds good?

File: bluebrain/repo-bluebrain/packages/coreneuron/package.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Coreneuron(CMakePackage):
     git      = "git@bbpgitlab.epfl.ch:hpc/coreneuron.git"
 
     version('develop', branch='master')
+    version('8.2.1', tag='8.2.1')
     version('8.2.0', tag='8.2.0')
     # 1.0.1 > 1.0.0.20210519 > 1.0 as far as Spack is concerned
     version('1.0.0.20220304', commit='2d08705')
```
Comment (sergiorg-hpc) [inline bluebrain/repo-bluebrain/packages/coreneuron/package.py:27]: @pramodk At some point, we should probably consider removing most of the temporary versions but one (e.g., latest). I find it silly that we still have these around, haha :P

Comment (pramodk) [reply bluebrain/repo-bluebrain/packages/coreneuron/package.py:27]: don't hesitate to remove those right away!

**Comment by olupton** [comment]
Can you also rebase? I still see `coreneuron@develop` in the logs: https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/340699#L1674

File: bluebrain/repo-bluebrain/packages/coreneuron/package.py [hunk 2]
```diff
@@ -104,7 +105,7 @@ class Coreneuron(CMakePackage):
     conflicts('+unified', when='~gpu')
 
     # Older versions do not support GPU execution in shared builds
-    conflicts('+shared', when='@:8.2.0 +gpu')
+    conflicts('+shared', when='@:8.9.9 +gpu')
```
Comment (olupton) [inline bluebrain/repo-bluebrain/packages/coreneuron/package.py]: This seems fine to me, this support will not be backported to the 8.2 series and probably not to any hypothetical 8.3 series either.

Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/coreneuron/package.py]:
```suggestion
    conflicts('+shared', when='@:8 +gpu')
```
More concise.

**Comment by olupton** [comment]
https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/341052#L1673 looks better, no more @develop.

**Review by olupton — APPROVED** [review]

**Pull request merged**